sys/netgraph/ng_pppoe.c

```diff
@@ -948,9 +949,10 @@ ng_pppoe_rcvmsg(node_p node, item_p item, hook_p lasthook)
 			neg->host_uniq.hdr.tag_type = PTT_HOST_UNIQ;
 			if (hulen == 0) {
 				/* Not provided, generate one */
-				neg->host_uniq.hdr.tag_len = htons(sizeof(sp));
-				bcopy(&sp, neg->host_uniq.data, sizeof(sp));
-				neg->host_uniq_len = sizeof(sp);
+				neg->host_uniq.hdr.tag_len = htons(sizeof(ptraddr_t));
+				*(ptraddr_t *)neg->host_uniq.data =
```

Comment:
Is this guaranteed aligned? Might need to keep it as a `bcopy` with a new temporary ptraddr_t.

Reply:
Changed back to a bcopy().  It's careful to copy it in the other direction as well in the existing code.